beancount/utils/test_utils.py

```diff
@@ -186,7 +186,7 @@ def new_function(self):
         allowed = ('buffering', 'encoding', 'newline', 'dir', 'prefix', 'suffix')
         if any(key not in allowed for key in kwargs):
             raise ValueError("Invalid kwarg to docfile_extra")
-        with tempfile.NamedTemporaryFile('w', **kwargs) as file:
+        with tempfile.NamedTemporaryFile('w', delete=False, **kwargs) as file:
```

Comment:
Who deletes the temporary file then?

Reply:
How is that possible that the files get deleted first?
Is it triggered by my call to flush()?

I agree, not deleting the files is horrible.

Reply:
@dnicolodi - I was curious and ran all the tests in Windows and surprisingly only 33 failures out of a +1300 test. Perhaps with a little effort we could support it. However I think there are other features that I need more (like AVERAGE booking) so probably it is just easier to install a Linux VM.

```
Ran 1355 tests in 36.826s

FAILED (failures=33, errors=29, skipped=22, expected failures=2)
Test failed: <unittest.runner.TextTestResult run=1355 errors=29 failures=33>
error: Test failed: <unittest.runner.TextTestResult run=1355 errors=29 failures=33>
```

@blais - I honestly don't know why Windows immediately deletes the file after the `with` block. It might be a garbage collection nuance since you pass the name of the file and not the file handler itself. It is dumb because this behavior defeats the very purpose of `NamedTemporaryFile`.

That being said we could drop this PR. 

Thanks for beancount :-)

Reply:
I looked into this a while ago: IIRC, the issue is not that the file is deleted, but that on Windows the file cannot be opened again while it is already opened for writing. The `delete=False` simply triggers a code path where the last file descriptor to the temporary file is dropped before the subsequent open.

I count 62 problematic tests (33 failures and 29 errors that do not happen when running the tests on a Unix platform). I think having the test suite work on Windows is a worthwhile goal, however, the proposed change is an ugly work-around. The issue needs to be fixed properly, ie without leaving temporary files around.

Reply:
Can't we just insert a random suffix to avoid collisions?

Reply:
Unfortunately, the issue is not of name collisions but of stupid Windows behavior regarding file handlers.